core/capabilities/registry.go

```diff
@@ -14,16 +14,40 @@ var (
 	ErrCapabilityAlreadyExists = errors.New("capability already exists")
 )
 
+type localRegistry interface {
+	GetLocalNode(ctx context.Context) (capabilities.Node, error)
+	ConfigForCapability(ctx context.Context, capabilityID string, donID uint32) (capabilities.CapabilityConfiguration, error)
+}
+
 // Registry is a struct for the registry of capabilities.
 // Registry is safe for concurrent use.
 type Registry struct {
-	lggr logger.Logger
-	m    map[string]capabilities.BaseCapability
-	mu   sync.RWMutex
+	localRegistry localRegistry
```

Comment:
Wdyt about renaming this? This Registry struct is also technically a local copy of the remote registry. Some options I thought of:
- metadata store
- metadata registry

 I came up with these names based on the fact that this struct stores a map from capabillity id -> a capability object, whereas localRegistry stores just plain data structures that describe DONs, node configs, and capability descriptors.